src/mount/conf.cc

```diff
@@ -53,9 +54,15 @@ extern "C" void mount_ceph_get_config_info(const char *config_file,
   for (const auto& mon : monc.monmap.addr_mons) {
     auto& eaddr = mon.first;
 
-    // For now, kernel client only accepts legacy addrs
-    if (!eaddr.is_legacy())
-      continue;
+    if (v2_addrs) {
+      // v2 addresses must support msgr2.
+      if (!eaddr.is_msgr2())
+	continue;
+    } else {
+      // v1 transports must be legacy
+      if (!eaddr.is_legacy())
+	continue;
+    }
 
```

Comment:
Nit: I think your old comment made more sense, as addresses by themselves don't "support" anything, especially given that we strip everything but ip and port below.  I'd amend it and leave it there, e.g.:
```
    /*
     * Filter v1 addrs if we're running in ms_mode=legacy. Filter
     * v2 addrs for any other ms_mode.
     */
```

Reply:
Done.